File: compiler/testData/loadJava/compiledJava/annotations/AnnotatedParameterInInnerClassConstructor.java
// SKIP_IN_FIR_TEST
// SKIP_IN_RUNTIME_TEST
package test;

import java.lang.annotation.ElementType;
import java.lang.annotation.Target;

class AnnotatedParameterInInnerClassConstructor {

    @Target(ElementType.TYPE_USE)
    public @interface Anno {
    }

    class InnerGeneric<T, K> {
    }

    Integer foo(InnerGeneric<@Anno String, InnerGeneric<String, @Anno InnerGeneric<@Anno String, String>>> a) { return 11; }
}

File: analysis/decompiled/decompiler-to-file-stubs/testData/clsFileStubBuilder/AnnotatedParameterInInnerClassConstructor/AnnotatedParameterInInnerClassConstructor.kt
// FIR_IDENTICAL
package test

annotation class AnnoA
annotation class AnnoB

class AnnotatedParameterInInnerClassConstructor {

    inner class Inner(@AnnoA a: String, @AnnoB b: String) {

    }

    inner class InnerGeneric<T>(@AnnoA a: T, @AnnoB
    b: String) {

    }
}